lib/cocoapods/external_sources/abstract_external_source.rb

```diff
@@ -141,17 +141,29 @@ def download_request
       # @return [void]
       #
       def store_podspec(sandbox, spec, json = false)
-        if spec.is_a? Pathname
-          spec = Specification.from_file(spec).to_pretty_json
-          json = true
-        elsif spec.is_a?(String) && !json
-          spec = Specification.from_string(spec, 'spec.podspec').to_pretty_json
-          json = true
-        elsif spec.is_a?(Specification)
-          spec = spec.to_pretty_json
-          json = true
+        case spec
+        when Pathname
+          spec = Specification.from_file(spec)
+        when String
+          path = "#{name}.podspec"
+          path << '.json' if json
+          spec = Specification.from_string(spec, path)
+        end
+        validate_podspec(spec)
+        sandbox.store_podspec(name, spec.to_pretty_json, true, true)
```

Comment:
I think I was asking because of the `json` parameter and the `path << '.json' if json` code. Is there still any point to that code?